cpp/src/Ice/Proxy.cpp

```diff
@@ -580,105 +574,3 @@ Ice::ObjectPrx::_twoway() const
         return _reference->changeMode(Reference::ModeTwoway);
     }
 }
-
-// TODO: move the code below to ProxyFunctions.cpp
-
-void
-IceInternal::throwNullProxyMarshalException(const char* file, int line, const Current& current)
-{
-    ostringstream os;
-    os << "null proxy passed to " << current.operation << " on object "
-       << current.adapter->getCommunicator()->identityToString(current.id);
-    throw MarshalException{file, line, os.str()};
-}
-
-namespace Ice
```

Comment:
This odd implementation of operator< and operator== (declared in ProxyFunctions.h) is because clang gets confused with the Ice::operator< and == templates in Comparable.h (= clang bug). Fixed that too.